### ISSUE TITLE
Can add issue to closed issue

### ISSUE THREAD

**Issue by ejouvin** [go-gitea/gitea#10779]
- Gitea version (or commit ref): 1.11.3
- Git version: 2.20.1
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
When I add an issue (Issue Child) as a dependency for another issue (Issue Parent), the parent issue is blocked by the child.
I can not close the parent until child is closed. That's fine.

But if Parent is already closed, I can add Child. And in that case we can see that Child is blocking Parent, but it is closed.
May be it works as designed.

...
